### ISSUE TITLE
Ignore InvalidPackageDeclaration in buildSrc folder

### ISSUE THREAD

**Issue by Mygod** [detekt/detekt#3946]
## Expected Behavior

Rule is not triggered.

## Observed Behavior

Rule is triggered.

## Steps to Reproduce

Run on this file: https://github.com/shadowsocks/shadowsocks-android/blob/f5bc88b83ebbd4d2941a65bea2eaa565f2072db4/buildSrc/src/main/kotlin/Helpers.kt#L1

## Context

This file should be in root package.

## Your Environment

* Version of detekt used: v1.17.1
* Version of Gradle used (if applicable): N/A
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version: Linux Mint 20
* Link to your project (if it's a public repository): See above

**Comment by marschwar** [comment]
This is sort of by design. Since detekt has no way of knowing what the source root is, any file that does not contain a package declaration is flagged. I see a few ways forward.

1. You can exclude files directly in `buildSrc/src/main/kotlin` from the rule (see below)
1. We could add a flag `ignoreMissingPackageDeclaration` to the rule
1.  We could ignore files in `src/*/java` and `src/*/kotlin` with a missing package declaration

```yaml
  InvalidPackageDeclaration:
    excludes: ['**/*.kts', '**/buildSrc/src/main/kotlin/*.kt']
```

To be honest I prefer option 1. :wink:

**Comment by BraisGabin** [comment]
As discussed in #3947 this is a platform specific customization that we, as detekt, doesn't want to have by default. There are users that doesn't use gradle.

But we are going to create MissingPackageDeclaration that should help with this: #3957
